### DISCUSSION TITLE
cypress coding dos and dont's

### DISCUSSION THREAD

**Discussion by aelbore** [cypress-io/cypress#21612] [Component Testing]
Should i provide a component function to cypress or should i do it in cypress like query the element?

for example: counter component
should i create an `increment` or `decrement` function in the component or should i querySelect the button the invoke the click event? 

in cypress

```typescript
cy.get("counter-element").find("button['data-qa-test-btn-increment']").click()
```

or 
```typescript
cy.get("counter-element").increment()
```

which is prepared to do?

**Comment by muratkeremozcan** [comment]
I'd kill that with `cy.getByCyLike('btn-increment')`.

https://github.com/muratkeremozcan/react-hooks-in-action-with-cypress/blob/main/cypress/support/commands.js#L35

I like the function too. Creating a function just for testing, hard-core, but probably not necessary.
